core/domain/suggestion_registry.py

```diff
@@ -878,10 +905,6 @@ def pre_update_validate(self, change):
             raise utils.ValidationError(
                 'The new change skill_id must be equal to %s' %
                 self.change.skill_id)
-        if self.change.question_dict == change.question_dict:
```

Comment:
Two things:

- Wouldn't you want to validate that at least one of the two fields have changed?
- Please include backend tests for the various cases (including the one you described where the difficulty changes but the question dict does not).


In general, validate things as fully as possible. The review comments I left were not a suggestion to make the validation weaker, but to make it correct.

Reply:
In that case, skill_difficulty is not an attribute of change. I am not quite sure how new skill_difficulty value should be passed into this function as pre_update_validate is implemented for general suggestions including questions as well as translations. But skill_difficulty is only an attribute of questions. Could you please help me to sort this one out? @seanlip

Reply:
I got an odd comment here by email but it seems to have disappeared. I'll reply anyway.

> In that case, skill_difficulty is not an attribute of change. I am not quite sure how new skill_difficulty value should be passed into this function as pre_update_validate is implemented for general suggestions including questions as well as translations. But skill_difficulty is only an attribute of questions. Could you please help me to sort this one out? @seanlip

Isn't skill_difficulty part of the change object? It seems to be, according to this: https://github.com/oppia/oppia/blob/b64813bbbda63d835ec9901cba16a2471f8e3a60/core/domain/question_domain.py#L115

Reply:
Yes, sorry I removed it since I found it. Apologies for not mentioning it. I will update it.

Reply:
Done